src/easyscience/Objects/new_variable/descriptor_number.py

```
@@ -193,8 +194,8 @@ def convert_unit(self, unit_str: str):
             raise TypeError(f'{unit_str=} must be a string representing a valid scipp unit')
         try:
             new_unit = sc.Unit(unit_str)
-        except Exception as message:
-            raise ValueError(message)
+        except UnitError as message:
+            raise UnitError(message) from None
```

What does the `from None` do?

---

Also used in other parts of the code

---

It seems that it suppresses the original error message. Whether or not we want this is up for discussion. Let's keep it as is for now.

---

It suppresses the full error path. I found the raised error to be much more readable without it.